report/4.Solution/SemanticsDescription.tex

```diff
@@ -0,0 +1,748 @@
+$t\ ::=\ str\ |\ x\ |\ t_1+t_2\ |\ t\ toString\ |\ (t)$\\
+$list\ ::=\ lst\ |\ x\ |\ list\ length\ |\ list\ @\ a\ |\ (list)$\\
+$tuple\ ::=\ tpl\ |\ x\ |\ tuple\ @\ a\ |\ (tuple)$\\
+$e\ ::=\ a\ |\ b\ |\ t\ |\ list\ |\ tuple\ |\ F_{exp}\ | L$\\
+$pOrf\ ::=\ f\ |\ p$\\
+$eOrL\ ::=\ e\ |\ L$\\
+$F_{exp}\ ::=\ f\ |\ eOrL\ f\ eOrL^*$\\
+$L\ ::= \{\ S\ \}\ |\ \{\ S\ return\ e\ \}\ |\ \{\ S;\ e\ \}$\\
+$S\ ::=\ x\ =\ e\ |\ S_1;S_2\ |\ b\ then\ L\ |\ L\ while\ b\ |\ p\ |\ eOrL\ p\ eOrL^*\ |\ D_V\ |\ D_P\ |\ \epsilon$\\
+$D_v\ ::=\ T\ x\ =\ e;\ S\ |\ T\ x\ = call_{exp};\ S\ |\ T\ x;\ D_v\ |\ \epsilon$\\
+$D_p\ ::=\ T\ pOrf\ =\ ([T\ x]\sim):\ T\ L;\ S\ |\ T\ pOrf;\ S\ |\ pOrf\ =\ ([T\ x]\sim):\ T\ L;\ S\ |\ \epsilon$
+
+\textbf{\large{Conventions}} \\
+For each variable $x$, an arbitrary variable-environment $env_V$ describes which address $adr$, $x$ is bound to.
+For simplicity, the set of all possible addresses, denoted $Adr$, is assumed to be equal to the set of all integers.
+\begin{center}
+	$Adr = \mathbb{Z}$
+\end{center}
+The set of all variable-environments, denoted \textbf{$Env_V$}, is the set of all partial functions from variables to addresses.
+\begin{center}
+	$Env_V = Var \cup \{next\} \rightharpoonup Adr$
+\end{center}
+For every $adr$, the function $new : Adr \rightarrow Adr$ returns the next $adr$, whether it is free or not.
+\begin{center}
+	$new (adr) = adr + 1$
+\end{center}
+A store $sto$ describes what values $v$ are pointed to by the addresses found in $Adr$.
+
+With the variable bindings found in $env_V$ and the address content found in $sto$, the expressions are capable of returning the values of the $sto$.
+This means that every transition of expressions must be on the form.
+\begin{center}
+	$env_V,sto \vdash e \rightarrow v$
+\end{center} 
+
+\subsection{Big-Step Semantic for Expressions}
+An expression is any syntactic component that evaluates to a value.
+In HCL most expressions behave the same, but a few transition-systems uses special notation.
+When this is the case, it will be made apparent before that transition-system.
+
+\textbf{\large{General Rules for Expressions}} \\
+All expression in the abstract syntax have two rewriting rules in common.
+The variable rule and the parenthesis rule.
+To avoid the possibility of cluttering the section, these rules are described below, using the metavariable $e \in Exp$, where the syntax category $Exp$ is the set of all expressions.
+
+\textbf{[$Parent_{BSS}$]}
+\begin{center}
+	\begin{math}
+	\cfrac
+		{env_V,\ sto \vdash e \rightarrow_e v}
+		{env_V,\ sto \vdash (e) \rightarrow_e v}
+	\end{math}
+\end{center}
+
+\textbf{[$Var_{BSS}$]}
+\begin{center}
+	\begin{math}
+	env_V,\ sto \vdash x \rightarrow_e v
+	\texttt{ if } env_V\ x = l
+	\texttt{ and } sto\ l = v
+	\end{math}
+\end{center}
+
+\textbf{\large{Arithmetic Expressions}}\\
+Arithmetic expressions returns a value $v$, where $v \in \mathbb{R}$ or $v \in \mathbb{R}$.
```

Comment:
>where $v \in \mathbb{R}$ or $v \in \mathbb{R}

you have an example where you say X or X.. maybe you meant something else?

Reply:
Made a new command that makes a proper cartasian product cross.
Fixed